pallene/parser.lua

```diff
@@ -502,16 +502,26 @@ function Parser:FuncStat(is_local)
 
     local fields = {}
     while self:try(".") do
```

Comment:
Since the grammar for toplevel functions is to only support names of kind `NAME("."NAME)?`, could this just be an `if self:try(".")` instead? Then we will no longer need the `if #fields == 1` check, since `field` could just be an NAME token or nil, instead of a table of NAME tokens.